### ISSUE TITLE
Could not find clang library during cmake

### ISSUE THREAD

**Issue by Safuya** [mozart/mozart2#258]
It seems cmake had an issue finding the clang library. What am I doing wrong? I'm running CentOS 7. I followed all of the steps to this point on the front page and had no issues. Now I get the below:
#############################################
[rhughes@localhost builds]$ mkdir mozart2-release
[rhughes@localhost builds]$ cd mozart2-release/
[rhughes@localhost mozart2-release]$ cmake -DCMAKE_BUILD_TYPE=Release ../../mozart2/
-- The CXX compiler identification is GNU 4.8.2
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Found Git: /usr/bin/git (found version "1.8.3.1") 
-- The C compiler identification is GNU 4.8.2
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Mozart 2.0.0-alpha.0+build.4138.3739933 (Tue, 28 Oct 2014 18:24:57 +0800)
-- Found Subversion: /usr/bin/svn (found version "1.7.14") 
CMake Error at vm/generator/main/CMakeLists.txt:15 (message):
  Could not find clang library in
  /home/rhughes/mozart_projects/builds/mozart2-release/externals/Build/llvm

-- Configuring incomplete, errors occurred!
[rhughes@localhost mozart2-release]$ ..
[rhughes@localhost builds]$ mozart2-release/
bash: mozart2-release/: Is a directory
[rhughes@localhost builds]$ cd mozart2-release/
[rhughes@localhost mozart2-release]$ cd externals/
[rhughes@localhost externals]$ ll
total 0
drwxrwxr-x. 3 rhughes rhughes 17 Mar 18 11:15 Build
drwxrwxr-x. 3 rhughes rhughes 17 Mar 18 11:15 Download
drwxrwxr-x. 3 rhughes rhughes 17 Mar 18 11:15 Install
drwxrwxr-x. 3 rhughes rhughes 17 Mar 18 11:15 Source
drwxrwxr-x. 3 rhughes rhughes 17 Mar 18 11:15 Stamp
drwxrwxr-x. 3 rhughes rhughes 17 Mar 18 11:15 tmp
[rhughes@localhost externals]$ cd Build/
[rhughes@localhost Build]$ ll
total 0
drwxrwxr-x. 2 rhughes rhughes 6 Mar 18 11:15 llvm
[rhughes@localhost Build]$ cd llvm
[rhughes@localhost llvm]$ ll
total 0
[rhughes@localhost llvm]$ la
total 0
drwxrwxr-x. 2 rhughes rhughes  6 Mar 18 11:15 .
drwxrwxr-x. 3 rhughes rhughes 17 Mar 18 11:15 ..
[rhughes@localhost llvm]$ 
#################################################
I have clang installed in /home/rhughes/mozart_projects/builds/llvm-release/bin. I've also added this to my paths.

**Comment by layus** [comment]
You should have a .../externals/llvm/tools/clang folder with a copy of clang sources.

Did you follow all the steps in the manual ? Especially
```
externals$ cd llvm/tools/
tools$ svn co http://llvm.org/svn/llvm-project/cfe/tags/RELEASE_34/final clang
```

**Comment by Safuya** [comment]
Hi. I believe I did, and when re-compiling llvm for the second time I saw messages for clang being installed. I also believe I managed to see the version of clang.
I've taken the easy route now though, removing all the source files and installing the virtual machine on VirtualBox instead while I'm plodding my way through Concepts, Techniques and Models of Computer Programming.

**Comment by layus** [comment]
Its good to hear that you found how to use mozart2.
You should alse be able to install the .rpm package located at http://sourceforge.net/projects/mozart-oz/files/v2.0.0-alpha.0/

**Comment by thezerobit** [comment]
I have the exact same problem, I followed all the instructions to the letter. I'm on Debian 8.0.

**Comment by thezerobit** [comment]
The documentation is wrong, the part it says you can skip (downloading and building gtest and llvm/clang) you can't skip, and you need to specify paths for them with cmake, the full command:

    cmake -DCMAKE_BUILD_TYPE=Release -DLLVM_SRC_DIR=../../externals/llvm/ -DLLVM_BUILD_DIR=../llvm-release/ -DGTEST_SRC_DIR=../../externals/gtest -DGTEST_BUILD_DIR=../gtest-debug ../../mozart2

**Comment by allentiak** [comment]
I have the same problem, but with gtest. I have also followed the instructions to the letter...
(I have even made a pull request for some documentation changes: see #269)

**Comment by allentiak** [comment]
I have just found a possible solution: explicitly state CLANG_SRC_DIR: See the end of the first comment in #257.
Closed in my pull request #269.
